### PR TITLE
Migrate to Jakarta Mail

### PR DESCRIPTION
**Pull request by basil**
Downstream of #166, #167, and jenkinsci/jenkins#6165. Bumps the baseline to one containing jenkinsci/jenkins#6165 and migrates to Jakarta Mail.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -88,6 +88,7 @@
         <dependency>
             <groupId>org.jenkins-ci.plugins</groupId>
             <artifactId>mailer</artifactId>
+            <version>403.v4da_0b_7596fa_6</version> <!-- TODO https://github.com/jenkinsci/mailer-plugin/pull/144 -->
```
Comment (jglick) [inline pom.xml:91]: https://github.com/jenkinsci/mailer-plugin/pull/144

**Comment by basil** [comment]
I think this should be released to avoid problems for people who have upgraded to Mailer 435.v79ef3972b_5c7 or later (which uses `jakarta.mail` imports).

**Comment by basil** [comment]
See [the release notes for Email Extension 2.90](https://github.com/jenkinsci/email-ext-plugin/releases/tag/email-ext-2.90) for suggested messaging.

**Comment by jglick** [comment]
> I think this should be released

Done: https://github.com/jenkinsci/workflow-basic-steps-plugin/releases/tag/980.v82219a_ed188e

> avoid problems for people who have upgraded to Mailer 435.v79ef3972b_5c7 or later

Is there some specific sort of problem you would foresee?

> [the release notes for Email Extension 2.90](https://github.com/jenkinsci/email-ext-plugin/releases/tag/email-ext-2.90)

That seems to be specific to people using custom in-process Groovy scripting, which would not apply here.

Maybe that the signature of `MimeMessageBuilder` changed incompatibly?

**Comment by basil** [comment]
> Is there some specific sort of problem you would foresee?

Yes, I suspect the `mail` step of Pipeline: Basic Steps 975.v8d89b_e9579b_b_ would stop working on 2.332.x once Mailer is upgraded to 435.v79ef3972b_5c7, because Pipeline: Basic Steps 975.v8d89b_e9579b_b_ uses JavaMail and consumes this library from Mailer, which (as of 435.v79ef3972b_5c7) is providing that library under new imports, which Pipeline: Basic Steps doesn't consume until 980.v82219a_ed188e.

> That seems to be specific to people using custom in-process Groovy scripting, which would not apply here.

What do you mean by "that"?

The "Jakarta imports" section is specific to people using custom Groovy scripting, which does not apply here.

The "Related plugin releases" section does apply here.

**Comment by jglick** [comment]
> The "Related plugin releases" section does apply here.

OK, I updated https://github.com/jenkinsci/mailer-plugin/releases/tag/435.v79ef3972b_5c7 and https://github.com/jenkinsci/workflow-basic-steps-plugin/releases/tag/980.v82219a_ed188e accordingly; can you check for accuracy please?

**Comment by basil** [comment]
Looks good to me

**Comment by jglick** [comment]
OK. I am preparing a `bom` update for this…

**Comment by basil** [comment]
Beat you to it in https://github.com/jenkinsci/bom/pull/1280

**Pull request merged**